src/pdftoraw.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2021 Thierry HUCHARD <thierry@ordissimo.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <unistd.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <gio/gio.h>
+#include <poppler.h>
+#include <gdk/gdk.h>
+
+#define DPI 150.0
+
+static struct option long_options[] = {
+    {"file",     required_argument, 0,  'f' },
+    {"pages",    no_argument,       0,  'p' },
+    {"test",     no_argument,       0,  't' },
+    {"width",    required_argument, 0,  'w' },
+    {"height",   required_argument, 0,  'W' },
+    {"raw",      required_argument, 0,  'r' },
+    {"help",     no_argument,       0,  'h' },
+    {0,          0,                 0,  0   }
+};
+
+static void
+print_usage(void) {
+    printf("Usage:\tpdftoraw  --filename=<filename>  --raw=<num-page>\n");
+    printf("\tpdftoraw  --filename=<filename>  --pages\n");
+    printf("\tpdftoraw  --filename=<filename>  --width=<num-page>\n");
+    printf("\tpdftoraw  --filename=<filename>  --height=<num-page>\n");
+    printf("\tpdftoraw  --help\n");
+}
+
+static int
+pdf_number_pages_get(PopplerDocument *doc)
+{
+    return poppler_document_get_n_pages(doc);
+}
+
+static unsigned char*
+pdf_page_raw_get(PopplerDocument *doc,
+                 int page_nr)
+{
+    cairo_surface_t *s = NULL;
+    cairo_t *cr = NULL;
+    unsigned char *data = NULL;
+    GdkPixbuf *pix = NULL;
+    double width = 0;
+    double height = 0;
+    int w = 0;
+    int h = 0;
+
+    PopplerPage *page = poppler_document_get_page(doc, page_nr);
```

Comment:
Formatting nitpick throughout this patch. Put spaces after function names:

`foo (bar);`

Reply:
fixed 3a6cd1e912004c2

Reply:
Still a lot of functions where this is missing a space.

Reply:
Fixed by de462ea